build-all.sh

```diff
@@ -78,26 +84,28 @@ while [ $# -gt 0 ]; do
     shift
 done
 if [ -z "$PREFIX" ]; then
-    echo "$0 [--enable-asserts] [--disable-dylib] [--full-llvm] [--with-python] [--disable-lldb] [--disable-lldb-mi] [--disable-clang-tools-extra] [--host=triple] [--with-default-win32-winnt=0x601] [--with-default-msvcrt=ucrt] [--enable-cfguard|--disable-cfguard] [--no-runtimes] [--no-tools] [--wipe-runtimes] dest"
+    echo "$0 [--host-clang[=clang]] [--enable-asserts] [--disable-dylib] [--full-llvm] [--with-python] [--disable-lldb] [--disable-lldb-mi] [--disable-clang-tools-extra] [--host=triple] [--with-default-win32-winnt=0x601] [--with-default-msvcrt=ucrt] [--enable-cfguard|--disable-cfguard] [--no-runtimes] [--no-tools] [--wipe-runtimes] dest"
     exit 1
 fi
 
-for dep in git cmake; do
+for dep in git cmake ${HOST_CLANG-}; do
     if ! command -v $dep >/dev/null; then
         echo "$dep not installed. Please install it and retry" 1>&2
         exit 1
     fi
 done
 
 if [ -z "$NO_TOOLS" ]; then
-    ./build-llvm.sh $PREFIX $LLVM_ARGS $HOST_ARGS
-    if [ -z "$NO_LLDB" ] && [ -z "$NO_LLDB_MI" ]; then
-        ./build-lldb-mi.sh $PREFIX $HOST_ARGS
-    fi
-    if [ -z "$FULL_LLVM" ]; then
-        ./strip-llvm.sh $PREFIX
+    if [ -z "${HOST_CLANG-}" ]; then
```

Comment:
I usually code shell scripts to work with `set -u` enabled, at which point the script will error out if you expand an unset variable. ties into the previous comment.

If you want a common style, I will adhere to it. Though I can only recommend aiming at making the scripts compatible with `set -u`.

Reply:
Nice, I wasn't aware of `set -u`. I can certainly see that it'd make some aspects of scripts more robust. (But then, I presume one should rather make sure that the variables always are set, rather than using `${FOO-}` for expansion.)

Off the top of my head, lots in these scripts use possibly-undefined variables, so I don't think I'll go ahead and rewrite the scripts to run with `set -u` at this point, but I see where you're coming from, thanks! Before finising the MR, I do appreciate matching the surrounding style.